### PR TITLE
Jason/issue 1127 4

### LINKED ISSUE #1127: [2.0 suggestion] AST refactoring: sort out wcsInfo, fullWCSInfor, wcsforTransformation...
so far we duplicate various AST objects such as wcsInfo, fullWCSInfor, wcsforConversion, so maybe we can sort out a way for clearer logic

- [x] init frame set with only at most 3D for fullWCSInfo & extracting spectral frame
- [x] extract 2D frame set from the frameset above
- [x] rearrange code in FrameStore

### PR DESCRIPTION
**Pull request by duidae**
closes #1127 

please help test the functions & values related wcs, thanks.

### CODE REVIEW

File: src/stores/FrameStore.ts
Comment (veggiesaurus) [inline src/stores/FrameStore.ts:737]: I wonder... does this still work with 3D cubes with a single channel? (i.e. `NAXIS = 3` and `NAXIS3 = 1`)

Comment (duidae) [reply src/stores/FrameStore.ts:737]: frameInfo.fileInfoExtended.depth > 1 is an existing old logic and spreading in quite some places, if it is incorrect, we can correct it in this PR

Comment (veggiesaurus) [reply src/stores/FrameStore.ts:737]: `frameInfo.fileInfoExtended.depth > 1` is generally correct for checking whether a cube is 2D or 3D, but when it comes to AST, we might still need to check things more carefully.

Comment (duidae) [reply src/stores/FrameStore.ts:737]: the original logic of 3D cube initialization is:
`if frameInfo.fileInfoExtended.depth > 1 then initFullWCS()`
and initFullWCS() gets NAXIS1~3 for AST

and current logic is just the same except adding a check for the spectral/stokes axis, so not sure the difference & have no clue about the checking,

@veggiesaurus would u mind elaborate that directly? thanks

Comment (veggiesaurus) [reply src/stores/FrameStore.ts:737]: sorry, I think you're right, it shouldn't make a difference. I was just worried that situations where you've got an image with NAXIS = 3 and NAXIS3 = 1 (which for CARTA's purposes is really a 2D image, just with a degenerate 3rd axis), AST wouldn't let you use `astTran2` since the frame info would be 3D.

**Review by veggiesaurus — APPROVED** [review]
changes look good. Ready to merge once @kswang1029 has tested and approved

**Comment by kswang1029** [comment]
@veggiesaurus @duidae will do by this week

**Review by kswang1029 — APPROVED** [review]
With some tests and comparisons with v1.4, I think the refactoring works. 👍

**Pull request merged**
